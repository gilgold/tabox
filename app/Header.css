.header {
	margin: 0 0 5px 0;
	background-color: var(--footer-bg-color);
	color: var(--bg-color);
	width: 100%;
	height: 40px;
	text-align: center;
	padding: 4px 0;
}

.avatar {
	height: 24px;
	width: 24px;
	margin: 4px 5px;
	border-radius: 12px;
	background-color: var(--avatar-bg);
}
.header_text {
	font-size: .75em;
	font-weight: 500;
	display: inline-block;
	color: var(--text-color);
	text-decoration: none;
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
}
.user_image {
	float: left;
	vertical-align: middle;
	display: flex;
	cursor: pointer;
}
.email {
	display: block;
	font-size: .62em;
	color: var(--text-color);
	text-align: left;
}