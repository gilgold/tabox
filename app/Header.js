import React, {useState, useEffect} from 'react'
import './Header.css';
import { 
    isLoggedInState, 
} from './atoms/globalAppSettingsState';
import { useRecoilState } from 'recoil';
import { browser } from '../static/index';
import { useSnackbar } from 'react-simple-snackbar';
import { SnackbarStyle } from './model/SnackbarTypes';

function LoginSection(props) {

    const [isLoggedIn, setIsLoggedIn] = useRecoilState(isLoggedInState);
    const [googleUser, setGoogleUser] = useState();
    const [openSnackbar, closeSnackbar] = useSnackbar({style: SnackbarStyle.SUCCESS});

    useEffect(async () => {
        if (isLoggedIn) {
            const {googleUser} = await browser.storage.local.get('googleUser');
            setGoogleUser(googleUser);
        }
    }, [isLoggedIn])
 
    const handleClick = async (e) => {
        if (isLoggedIn) {
            await props.logout();
            openSnackbar('Sync has been disabled', 3000)
        } else {
            await browser.runtime.sendMessage({type: 'login'}).then(async (response) => {
                setGoogleUser(response);
                setIsLoggedIn(true);
                openSnackbar('Sync is now enabled!', 3000)
                await props.applyDataFromServer();
            })
        }
    }

    return <div className="user_image" title={`Click here to ${isLoggedIn && googleUser ? 'disable' : 'enable'} Google Drive sync`} onClick={handleClick}>
                <div className="row">
                    <div className="column">
                        <div className="row double-row">
                            <img id="avatar" className="avatar" src={ isLoggedIn && googleUser && googleUser.photoLink ? googleUser.photoLink : '/images/not_signed_in.png' } alt="user avatar" />
                        </div>
                    </div>
                    { isLoggedIn && googleUser ? (
                        <div className="column">
                            <div className="row header_text">
                                Sync enabled for {googleUser.displayName}
                            </div>
                            <div className="row email">
                                {googleUser.emailAddress}
                            </div>
                        </div>
                    ) : <span className="header_text">Signin with Google to enable sync</span> }
                </div>
            </div>;
}

function Header(props) {
  return <header className="header">
            <LoginSection applyDataFromServer={props.applyDataFromServer} logout={props.logout}/>
        </header>;
};

export default Header;