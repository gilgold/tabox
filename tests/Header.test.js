import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../app/Header';
import { RecoilRoot } from 'recoil';

describe('Header -- Sync disabled', () => {
  test('Header renders correctly - sync disabled', () => {
    const component = renderer.create(
      <RecoilRoot>
        <Header />
      </RecoilRoot>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});