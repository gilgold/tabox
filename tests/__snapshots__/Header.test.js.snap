// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header -- Sync disabled Header renders correctly - sync disabled 1`] = `
<header
  className="header"
>
  <div
    className="user_image"
  >
    <div
      className="row"
      onClick={[Function]}
      title="Click here to enable Google Drive sync"
    >
      <div
        className="column"
      >
        <div
          className="row double-row"
        >
          <img
            alt="user avatar"
            className="avatar"
            id="avatar"
            src="/images/not_signed_in.png"
          />
        </div>
      </div>
      <span
        className="header_text"
      >
        Signin with Google to enable sync
      </span>
    </div>
  </div>
  <div
    className="settings-wrapper"
  >
    <div
      className="settings-button"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <svg
        color="var(--text-color)"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "var(--text-color)",
          }
        }
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M2.132 13.63a9.942 9.942 0 0 1 0-3.26c1.102.026 2.092-.502 2.477-1.431.385-.93.058-2.004-.74-2.763a9.942 9.942 0 0 1 2.306-2.307c.76.798 1.834 1.125 2.764.74.93-.385 1.457-1.376 1.43-2.477a9.942 9.942 0 0 1 3.262 0c-.027 1.102.501 2.092 1.43 2.477.93.385 2.004.058 2.763-.74a9.942 9.942 0 0 1 2.307 2.306c-.798.76-1.125 1.834-.74 2.764.385.93 1.376 1.457 2.477 1.43a9.942 9.942 0 0 1 0 3.262c-1.102-.027-2.092.501-2.477 1.43-.385.93-.058 2.004.74 2.763a9.942 9.942 0 0 1-2.306 2.307c-.76-.798-1.834-1.125-2.764-.74-.93.385-1.457 1.376-1.43 2.477a9.942 9.942 0 0 1-3.262 0c.027-1.102-.501-2.092-1.43-2.477-.93-.385-2.004-.058-2.763.74a9.942 9.942 0 0 1-2.307-2.306c.798-.76 1.125-1.834.74-2.764-.385-.93-1.376-1.457-2.477-1.43zM12 15a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
          />
        </g>
      </svg>
    </div>
    <div
      className="szh-menu-container settings-items-wrapper"
      onBlur={[Function]}
      onKeyDown={[Function]}
    />
  </div>
</header>
`;
